packages/contractkit/src/wallets/rpc-wallet.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  ensureLeading0x,
+  normalizeAddressWith0x,
+  privateKeyToAddress,
+} from '@celo/utils/lib/address'
+// @ts-ignore-next-line
+import { account as Account } from 'eth-lib'
+import * as ethUtil from 'ethereumjs-util'
+import { EncodedTransaction, provider, Tx } from 'web3-core'
+import { DefaultRpcCaller, RpcCaller } from '../utils/rpc-caller'
+import { EIP712TypedData } from '../utils/sign-typed-data-utils'
+import { encodeTransaction, rlpEncodedTx } from '../utils/signing-utils'
+import { Wallet } from './wallet'
+
+enum RpcMethod {
+  addPersonal = 'personal_importRawKey',
+  unlockPersonal = 'personal_unlockAccount',
+  signPersonal = 'personal_sign',
+  signTransaction = 'eth_signTransaction',
+  signTypedData = 'eth_signTypedData',
+}
+
+const currentTimeInSeconds = () => Math.round(Date.now() / 1000)
+
+export class RpcWallet implements Wallet {
```

Comment:
I prefer that you follow the same pattern that we already have in the contractKit.

The `Wallet` is the one that store the account info. And the `Signer` the one that know how to sign. This made easier every test, and will make all the wallets extend from the same base. Otherwise will depend on the implementation and that will be hard to follow in the future.

This could be a case of `RemoteWallet`, because the RPC call depends on the provider, and this wallet could allow to call a remote node.